quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -1273,7 +1277,19 @@ export class CompilerVisitor implements IRVisitor {
         // as evaluation was not successful.
         this.recoverNextVars(savedValues)
         this.resetTrace(just(rv.mkList(savedTrace)))
-        break
+
+        if (kind === 'then' && nactionsLeft > 0 && result.isJust() && !(result.value as RuntimeValue).toBool()) {
+          // Cannot extend a run. Emit an error message.
+          const actionNo = actions.length - (nactionsLeft + 1)
+          this.errorTracker.addRuntimeError(
+            actionId(actionNo),
```

Comment:
The use of `actionId` doesn't make sense to me. If we need to be fetching the id based on the action number, why is the ID hard-coded when used on line 1344?

Reply:
When we execute `A.then(B)`, we want the error listener to highlight `A`. In case of `reps` below, we have multiple copies of `A`. This is why all occurences of `A` have the same id.

Reply:
Maybe add an explanatory comment?